### ISSUE TITLE
Regarding the optimal configuration of memory usage

### ISSUE THREAD

**Issue by xumingyu2021** [erogol/BlaGPT#2]
I have try "torchrun --standalone --nproc_per_node=8 train.py --run_name pre_post_norm --model_name blagpt",
and the  memory_usage is around 42,000M , while the memory_usage of Best Model So Far is around 49,000M. 

I am curious about what operation caused such a significant increase in memory usage.

**Comment by xumingyu2021** [comment]
Besides, I also want to know how the learning rate is determined. Because it seems that the default lr in train_new.py and train.py is different? And I tried to increase LR from 1e-3 to 2e-3, the val-loss of KV shifting attention is 3.2891. 
https://github.com/xumingyu2021/BlaGPT/blob/main/bla_gpt/logs/kvshifting_lr2e-3_0.txt.

**Comment by erogol** [comment]
Sorry for a very late response. 

About mem usage, I've not run much testing for mem usage. so I don't know why. But happy to dive deeper if you have any pointers. 

About LR, I did not really optimize. And yes, you can definitely find better lr for each configuration. I added a lr finder but it did not bring much improvement for an epoch of training.
